doc/stack_vs_cabal_config.md

```diff
@@ -0,0 +1,135 @@
+<div class="hidden-warning"><a href="https://docs.haskellstack.org/"><img src="https://rawgit.com/commercialhaskell/stack/master/doc/img/hidden-warning.svg"></a></div>
+
+# stack vs cabal config files
+
+Due to their apparent overlap, the purpose of the following three files can be
+unclear:
+
+* `stack.yaml`
+* A cabal file, e.g. `my-package.cabal`
+* `package.yaml`
+
+The last two are easy to explain: `package.yaml` is a file format supported by
+[hpack](https://github.com/sol/hpack#readme), which adds some niceties on top
+of cabal like YAML syntax support and automatic generation of `exposed-modules`
+lists. However, it's just a frontend to cabal files. So for this document,
+we're instead going to focus on the first two and try to answer:
+
+_What's the difference between a `stack.yaml` file and a cabal file?_
+
+## Package versus project
+
+Cabal is a build system, which is used by Stack. Cabal defines the concept of a
+_package_. A package has:
+
+* A name and version
+* 0 or 1 libraries
```

Comment:
There should be at least one library or one executable. Also more correct term would be *exposed libraries*, as `Cabal` does support internal libraries.

Reply:
This is a level of pedantry not needed for understanding this topic.

Reply:
If you aren't pedantic, people *will* misunderstand or misinterpret. I know it's hard to be terse and exact at the same time, but if the goal is to reduce beginner's confusion, we all should pay attention to the docs.

Reply:
OK, but my statement is still _true_, just _incomplete_. I also haven't mentioned that a package needs to specify a build-type or a bunch of other things necessary. What concrete confusion do you expect users to encounter here?